homeassistant/components/atag/climate.py

```diff
@@ -0,0 +1,124 @@
+"""Initialization of ATAG One climate platform."""
+from typing import List, Optional
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    PRESET_AWAY,
+    PRESET_BOOST,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from . import CLIMATE, DOMAIN, ENTITY_TYPES, AtagEntity
+
+PRESET_SCHEDULE = "Auto"
+PRESET_MANUAL = "Manual"
+PRESET_EXTEND = "Extend"
+SUPPORT_PRESET = [
+    PRESET_MANUAL,
+    PRESET_SCHEDULE,
+    PRESET_EXTEND,
+    PRESET_AWAY,
+    PRESET_BOOST,
+]
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+HVAC_MODES = [HVAC_MODE_AUTO, HVAC_MODE_HEAT, HVAC_MODE_OFF]
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Load a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities([AtagThermostat(coordinator, ENTITY_TYPES[CLIMATE])])
+
+
+class AtagThermostat(AtagEntity, ClimateDevice, RestoreEntity):
+    """Atag climate device."""
+
+    def __init__(self, coordinator, atagtype):
+        """Initialize with fake on/off state."""
+        self._on = None
+        super().__init__(coordinator, atagtype)
+
+    async def async_added_to_hass(self):
+        """Register callbacks & state restore for fake "Off" mode."""
+        await super().async_added_to_hass()
+        last_state = await self.async_get_last_state()
+        if last_state:
+            self._on = last_state.state != HVAC_MODE_OFF
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def hvac_mode(self) -> Optional[str]:
+        """Return hvac operation ie. heat, cool mode."""
+        if not self._on:
+            return HVAC_MODE_OFF
+        if self.coordinator.atag.hvac_mode in HVAC_MODES:
+            return self.coordinator.atag.hvac_mode
```

Comment:
All exits should return explicitly if any exit return explicitly.
```suggestion
            return self.coordinator.atag.hvac_mode
        return None
```

Reply:
👍